base_layer/core/src/validation/header/header_full_validator.rs

```diff
@@ -216,5 +216,9 @@ fn check_pow_data(block_header: &BlockHeader, consensus_constants: &ConsensusCon
             }
             Ok(())
         },
+        PowAlgorithm::Cuckaroo => {
+            // Cuckaroo checks this specifically.
```

Comment:
we can add the length check here, then we dont have to do it in the cuckaroo and we have all the length checks in the same place